clang/lib/StaticAnalyzer/Checkers/MallocChecker.cpp

```diff
@@ -1877,16 +1923,43 @@ static bool didPreviousFreeFail(ProgramStateRef State,
   return false;
 }
 
+static void printOwnershipTakesList(raw_ostream &os, CheckerContext &C,
+                                    const Expr *E) {
+  if (const CallExpr *CE = dyn_cast<CallExpr>(E)) {
+    const FunctionDecl *FD = CE->getDirectCallee();
+    if (!FD)
+      return;
+
+    if (!FD->hasAttrs())
+      return;
+
+    // Only one ownership_takes attribute is allowed
+    for (const auto *I : FD->specific_attrs<OwnershipAttr>()) {
+      OwnershipAttr::OwnershipKind OwnKind = I->getOwnKind();
+
+      if (OwnKind != OwnershipAttr::Takes)
+        continue;
+
+      os << ", which takes ownership of " << '\'' << I->getModule()->getName()
+         << '\'';
```

Comment:
```suggestion
      os << ", which takes ownership of '" << I->getModule()->getName() << '\'';
```
This way it would even fit into a single line.